### ISSUE TITLE
eos_cli_config_gen add support for interface "speed auto"

### ISSUE THREAD

**Issue by c-po** [aristanetworks/ansible-avd#319]
**Issue Type**
- Template enhancement

As of now it is not possible to change the advertised autonegotiation modes via ansible-avd. Given the following implementation: https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2#L12-L14 one can either choose between a fully "fixed" configuration or "autoneg". There is some equipment out there which requires autoneg with a capability of 100 MBit fdx which can't be set at this time.

**Is your feature request related to a problem? Please describe.**
Networking gear which required 100 MBit autoneg fdx can't be configured

**Describe the solution you'd like**
Template extension to choose between forced and auto option form the EOS CLI

**Describe alternatives you've considered**
A configlet seems not to be the perfect fit here :(

We should discuss a change of the data model how this feature can be represented in a backwards compatible way. After the data-model is fixed I can submit a PR

## Proposed data-model
```
# Routed Interfaces
ethernet_interfaces:
  <Ethernet_interface_1 >:
    description: < description >
    shutdown: < true | false >
    speed: < forced_interface_speed >
      auto: < autoneg_offered_interface_speed >
```

Thus already existing YML definitions won't break. @aphillipps what you think?

**Comment by aphillipps** [comment]
Here is my suggestion as there are multiple ways of configuring an interface in EOS. Note that this will unfortunately break the current model but we would at least cover all the cases. Please, let me know if I missed any and if you see any issue with this model.

```yaml
speed:
      mode: < no-auto | forced | auto >
      speed: < interface_speed >
```
Example1:
```yaml
    speed:
      mode: auto
      speed:
```
Renders:
```cli
interface Ethernet1
   speed auto
```
Example2:
```yaml
    speed:
      mode: auto
      speed: 100full
```
Renders:
```cli
interface Ethernet2
   speed auto 100full
```
Example3:
```yaml
    speed:
      mode: forced
      speed: 100full
```
Renders:
```cli
interface Ethernet3
   speed forced 100full
```
Example4:
```yaml
    speed:
      mode: no-auto
      speed: 100full
```
Renders:
```cli
interface Ethernet4
   speed 100full
```

**Comment by c-po** [comment]
Hi @aphillipps,

if breaking the backwards compatibility is okay for you I'm happy to implement this as you outlined above. Those fits into my idea of how it could have been done by not keeping it backwards compatible.

**Comment by carlbuchmann** [comment]
@aphillipps @c-po , I agree with the proposed data model by @aphillipps. Since it will break the data model, I'm tagging this for 1.2.0 milestone, and the associated PR will only get merged to devel once we release 1.1.1. You will have to update the molecule tests accordingly. Note, we are currently in the process of refactoring the testing for eos_cli_config_gen, so I will let you know when this is ready.
